src/main/java/org/folio/consortia/service/impl/SharingSettingServiceImpl.java

```diff
@@ -113,6 +109,42 @@ private UUID publishRequest(UUID consortiumId, PublicationRequest publicationReq
     return null;
   }
 
+  private void checkEqualsOfPayloadIdWithSettingId(SharingSettingRequest sharingSettingRequest) {
+    String sharingSettingId =String.valueOf(sharingSettingRequest.getSettingId());
+    String payloadId = getPayloadId(sharingSettingRequest.getPayload());
+    if (ObjectUtils.notEqual(sharingSettingId, payloadId)) {
+      throw new IllegalArgumentException("id in payload is not equal to settingId");
+    }
+  }
+
+  private String getPayloadId(Object payload) {
+    JsonNode payloadNode = objectMapper.convertValue(payload, JsonNode.class);
+    return payloadNode.get("id").asText();
+  }
+
+  private PublicationRequest createPublicationRequestForSetting(SharingSettingRequest sharingSettingRequest, String httpMethod) {
+    PublicationRequest publicationRequest = new PublicationRequest();
+    publicationRequest.setMethod(httpMethod);
+    String url = sharingSettingRequest.getUrl();
+    String[] parts = url.split("/");
+    String lastPart =  parts[parts.length - 1];
+    Pattern uuidRegex = Pattern.compile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$");
```

Comment:
Can this be simplified without using regexp, so if PUT requires - we just add '/' + {{id}} at the end ?

Reply:
UI will always populate url like for POST requests, because they don't know in each tenants we need to create setting and in each - to update.
So we can simplify our BE code

Reply:
Ok, then I will simply it. are we sure every time we expect correct URL, or should I add some URL validation

Reply:
I did not think about it before, but yes, it will be good if we add such validation in both sahring setting and in publication service. You can use UriComponents from spring or Apache Commons Validator

Reply:
/organization/ - This URL was considered valid by UrlValidator of apache common but this URL is not work because it will throw error not found module 

So created new regex pattern currently it is working good added some tests